### PR TITLE
☸️ improved verse navigation

### PR DESCRIPTION
**Pull request by daybologic**
Improved navigation

### Summary by CodeRabbit

Based on the summary of changes, here are my suggestions:

1. `data/static/browsing_head.html, data/static/verse.html`: Ensure that the new HTML structure is semantic and accessible. Use ARIA roles where necessary to improve accessibility. Also, ensure that the URLs used for navigation are SEO-friendly.

2. `lib/Chleb/Bible/Book.pm, lib/Chleb/Bible/Chapter.pm`: The `getPath` method should handle any potential errors or exceptions when joining the short name. Consider edge cases such as null or empty values.

3. `lib/Chleb/Utils.pm, t/Utils_colorIndexFromWord.t, t/Utils_colorIndexFromWord.t`: When testing the `colorIndexFromWord` function, make sure to cover all possible edge cases. This includes not only valid inputs but also invalid ones. Also, consider performance implications of the tiny rolling hash algorithm, especially for large inputs.

4. `data/tests/1/lookup HTML, redirect.sh`: In the Bash script, ensure proper error handling. If fetching headers fails or if the results do not match expected values, the script should handle these situations gracefully. 

5. `lib/Chleb/Server/Dancer2.pm`: Be careful with changes related to UTF-8 encoding. Improper handling of character encoding can lead to bugs that are difficult to diagnose. Also, changing function calls from `__param` to `_param` might break existing functionality if not done carefully. Make sure to test thoroughly after making these changes.

### CODE REVIEW

File: data/static/public/style.css [hunk 1]
```diff
@@ -8,6 +8,145 @@ body {
...
+.tag-color-37 { background-color: hsl(209, 65%, 40%); }
+.tag-color-38 { background-color: hsl(214, 65%, 40%); }
+.tag-color-39 { background-color: hsl(220, 65%, 40%); }
+.tag-color-40 { background-color: hsl(226, 65%, 40%); }
+.tag-color-41 { background-color: hsl(231, 65%, 40%); }
+.tag-color-42 { background-color: hsl(237, 65%, 40%); }
+.tag-color-43 { background-color: hsl(243, 65%, 40%); }
+.tag-color-44 { background-color: hsl(248, 65%, 40%); }
+.tag-color-45 { background-color: hsl(254, 65%, 40%); }
+.tag-color-46 { background-color: hsl(260, 65%, 40%); }
```
Comment (daybologic) [inline data/static/public/style.css:66]: I don't think we want to use Javascript if we can help it.  This system should work on Lynx and elinks!  Also, there is likely no need to generate these colors.

File: t/Utils_colorIndexFromWord.t [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+sub testFixedMappings {
+	my ($self) = @_;
+	plan tests => 3;
+
+	is_deeply(Chleb::Utils::colorIndexFromWord('rebuke'), 30, "'rebuke' -> 30");
+	is_deeply(Chleb::Utils::colorIndexFromWord('warning'), 22, "'warning' -> 22");
+	is_deeply(Chleb::Utils::colorIndexFromWord('joy'), 50, "'joy' -> 50");
+
+	return EXIT_SUCCESS;
+}
```
[the remark this answers is not in the document]
Comment (daybologic) [reply t/Utils_colorIndexFromWord.t:55]: f526c5c

File: t/Utils_colorIndexFromWord.t [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			plan tests => 2;
+
+			my $index = Chleb::Utils::colorIndexFromWord($word);
+			cmp_ok($index, '>=', 0, 'at least zero');
+			cmp_ok($index, '<=', 63, 'no more than 63');
+		};
+	}
+
+	return EXIT_SUCCESS;
+}
```
[the remark this answers is not in the document]
Comment (daybologic) [reply t/Utils_colorIndexFromWord.t:88]: f526c5c

File: lib/Chleb/Server/Moose.pm [hunk 4]
```diff
@@ -859,53 +859,114 @@ sub __verseToHtml {
...
+	my $random;
+	{
+		my $pattern = "<a class=\"vn-link vn-chapter\" href=\"%s\">%s</a>";
+		if ($function == $FUNCTION_RANDOM) {
+			#$random = sprintf($pattern, $json->[0]->{links}->{self}, 'another'); # FIXME: because testament is '' which is invalid
+			$random = sprintf($pattern, '/1/random', 'another'); # FIXME: use self link, once it works
+		} else {
+			$random = sprintf($pattern, '/1/random', 'random');
+		}
 	}
```
Comment (daybologic) [inline lib/Chleb/Server/Moose.pm:873]: mis-reading the diff?

File: t/Utils_colorIndexFromWord.t [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+__PACKAGE__->meta->make_immutable;
+
+package main;
+use strict;
+use warnings;
+
+exit(UtilsColorIndexFromWordTests->new->run());
```
[the remark this answers is not in the document]
Comment (daybologic) [reply t/Utils_colorIndexFromWord.t:96]: handled by superclass Test::Module::Runnable's run()

File: lib/Chleb/Utils.pm [hunk 6]
```diff
@@ -369,6 +369,26 @@ sub explodeHtmlFilePath {
...
+
+	my $h = 0;
+
+	# Tiny rolling hash (inspired by djb2-style hash)
+	foreach my $ch (split(m//, $word)) {
+		$h = (($h * 33) + ord($ch)) & 0xFFFFFFFF; # keep it in 32 bits
+	}
+
+	return $h & 63; # low 6 bits => 0..63
+}
```
[the remark this answers is not in the document]
Comment (daybologic) [reply lib/Chleb/Utils.pm:390]: The words are only English words from the JSON-side, which came from ChatGPT, not an open-set of user-input; it is detainted, so not a concern.  undef is already handled via the //= operator

File: data/static/public/style.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (daybologic) [reply data/static/public/style.css:66]: Considered, but it probably isn't worth is for this use-case.

File: lib/Chleb/Server/Moose.pm [hunk 7]
```diff
@@ -394,7 +394,7 @@ sub __votd {
 		}
 
 		if ($contentType eq $Chleb::Server::MediaType::CONTENT_TYPE_HTML) { # text/html
-			return __verseToHtml(\@json, $FUNCTION_VOTD);
+			return __verseToHtml($verse, \@json, $FUNCTION_VOTD);
```
[the remark this answers is not in the document]
Comment (daybologic) [reply lib/Chleb/Server/Moose.pm]:
```
[241019Z DEC 25][palmer@cue] chleb-bible-search feature/openai-sentiment-html/1 $ grep -rn __verseToHtml lib
lib/Chleb/Server/Moose.pm:258:          return $self->__verseToHtml(\@verse, \@json, $FUNCTION_LOOKUP);
lib/Chleb/Server/Moose.pm:312:                  return $self->__verseToHtml($verse, \@json, $FUNCTION_RANDOM);
lib/Chleb/Server/Moose.pm:327:          return $self->__verseToHtml($verse, [$json], $FUNCTION_RANDOM);
lib/Chleb/Server/Moose.pm:399:                  return $self->__verseToHtml($verse, \@json, $FUNCTION_VOTD);
lib/Chleb/Server/Moose.pm:849:sub __verseToHtml {

File: lib/Chleb/Server/Moose.pm [hunk 8]
```diff
@@ -859,53 +859,114 @@ sub __verseToHtml {
...
 
-	if ($verseCount == 1) {
-		$output .= sprintf(" [%s]\r\n", $translation);
-	} else {
-		$output .= sprintf("\r\n\r\n\t(%s)\r\n", $translation);
+	my $sentiments = '';
+	foreach my $sentiment ($emotion, @allTones) {
+		my $colorIndex = Chleb::Utils::colorIndexFromWord($sentiment);
+		$sentiments .= "<span class=\"tag tag-color-${colorIndex}\">$sentiment</span> ";
 	}
```
[the remark this answers is not in the document]
Comment (daybologic) [reply lib/Chleb/Server/Moose.pm:913]: Will change when required -- agile.
